patroni/psycopg.py

```diff
@@ -14,27 +22,80 @@
     except ImportError:
         _legacy = True
 
-    def quote_literal(value, conn=None):
+    if TYPE_CHECKING:
+        from psycopg2.extensions import connection
+
+    def quote_literal(value: Any, conn: Optional["connection"] = None) -> str:
+        """Quote *value* as a SQL literal.
+
+        .. note::
+            *value* is quoted through ``psycopg`` adapters.
+
+        :param value: value to be quoted.
+        :param conn: if a connection is given then :func:`quote_literal` checks if any special handling based on server
+            parameters needs to be applied to *value* before quoting it as a SQL literal.
+
+        :returns: *value* quoted as a SQL literal.
+        """
         value = adapt(value)
         if conn:
             value.prepare(conn)
         return value.getquoted().decode('utf-8')
 except ImportError:
-    from psycopg import connect as __connect, sql, Error, DatabaseError, OperationalError, ProgrammingError
+    from psycopg import connect as __connect, sql, Error, DatabaseError, OperationalError, ProgrammingError, Connection
+
+    def _connect(*args: Any, **kwargs: Any) -> Connection[Any]:
+        """Call ``psycopg.connect`` with ``*args`` and ``**kwargs``.
```

Comment:
```suggestion
    def _connect(dsn: str, **kwargs: Any) -> Connection[Any]:
        """Call ``psycopg.connect`` with ``dsn`` and ``**kwargs``.
```

Reply:
By using `dsn` instead of `*args`, behave test start to fail because they don't use `dsn` parameter:

```
        File "features/environment.py", line 283, in _connection
          self._conn = psycopg.connect(**self._connkwargs)
        File "/home/runner/work/patroni/patroni/patroni/psycopg.py", line 100, in connect
          ret = _connect(*args, **kwargs)
      TypeError: _connect() missing 1 required positional argument: 'dsn'
  ```
  
  I attempted a few different options to overcome that issue, but was not able yet.
  
  I wonder if we should just acknowledge that pyright issue?

Reply:
Sorry, my bad. I didn't try it with behave tests :(
`psycopg.connect()` expects `conninfo` doesn't like None. Setting dsn to en empty string fixes it:
```python
    def _connect(dsn: str = "", **kwargs: Any) -> Any:
```

Reply:
I can't believe I have not tried with an empty string 😩 . I attempted with `None`, but not with an empty string. Thanks for that!